### ISSUE TITLE
How to prevent same start and end date selection

### ISSUE THREAD

**Issue by Saif03** [SaturnTeam/saturn-datepicker#98]
Is there anyway to configure minimum span in between selected date range? e,g, for my use case start and end dates must be at least one day apart, so that users won't be able to select same day as both start and end date.

similarly though I don't exactly need the opposite of this, but it will be cool if there is some way to configure maximum allowed span for date ranges? e.g. at maximum the number of days between start and end dates can be 7 days.

If this is something not currently supported but can be added quickly, then perhaps someone can point me in the direction where this can be configured so I can open a PR.

**Comment by SaturnTeam** [comment]
Sorry. You can only validate after selection

**Comment by Saif03** [comment]
@SaturnTeam I have tried adding validation on `dateChange` event;

```
    validateRange() {
    const range = this.form.value.date;
    if (range) {
      const startDate = range.begin;
      const endDate = moment(range.end);
      if (moment(startDate).isSame(endDate, 'day')) {
        const newValue = { begin: startDate, end: endDate.add(1, 'day').toDate() };
        this.form.patchValue({ date: newValue });
      }
    }
  }
```
doing this updates the placeholder though but on opening the calendar it still shows the old date,

![Screenshot (44)](https://user-images.githubusercontent.com/15065985/62157301-43d9ff00-b326-11e9-8be2-de39712a9936.png)

You can check this [StackBlitz](https://stackblitz.com/edit/angular-vn9vfb?file=src%2Fapp%2Fapp.component.ts) for further details.

**Comment by Saif03** [comment]
@SaturnTeam do you have any update regarding my last comment?

**Comment by JamesHearts** [comment]
It would be useful to put a "min" and "max" ranges on the date picker. For example maybe the user can not pick a range that is less than 7 days.

**Comment by Saif03** [comment]
@JamesHearts exactly, and for inspiration this JQuery based [date range picker ](https://github.com/dangrossman/daterangepicker) supports configuring properties `maxSpan` and `isInvalidDate` which allows wiring similar behavior e.g.

`maxSpan: { years: 2, months: 1 }`

@SaturnTeam in the mean time is there any alternative way I can configure this at my end? can you have a look at my comment above?

**Comment by SaturnTeam** [comment]
@Saif03 Datepicker supports custom validation, but not on time when you selecting the second date.
